sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts

```diff
@@ -1,100 +1,54 @@
+import { Span, CanonicalCode } from "@opentelemetry/api";
 
-import { RestError } from "@azure/core-http";
+import { RestError, OperationOptions } from "@azure/core-http";
+import { createSpanFunction } from "@azure/core-tracing";
+import { AppConfigurationClient } from "../appConfigurationClient";
 
-/**
- * @internal
- */
-export interface Spannable {
-  spanOptions?: SpanOptions;
-}
+/** @internal */
+export const createSpan = createSpanFunction({
+  namespace: "Microsoft.AppConfiguration",
+  packagePrefix: "Azure.Data.AppConfiguration"
+});
 
 /**
+ * Traces an operation and properly handles reporting start, end and errors for a given span
+ *
+ * @param operationName - Name of a method in the TClient type
+ * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
+ * @param fn - The function to call with an options class that properly propagates the span context
+ *
  * @internal
  */
-export class Spanner<TClient> {
-  constructor(private baseOperationName: string) {}
-
-  /**
-   * Traces an operation and properly handles reporting start, end and errors for a given span
-   *
-   * @param operationName - Name of a method in the TClient type
-   * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
-   * @param fn - The function to call with an options class that properly propagates the span context
-   * @param translateToCanonicalCodeFn - An optional function to translate thrown errors into a CanonicalCode for the span
-   */
-  async trace<OptionsT extends Spannable, ReturnT>(
-    operationName: keyof TClient,
-    options: OptionsT,
-    fn: (options: OptionsT, span: Span) => Promise<ReturnT>,
-    translateToCanonicalCodeFn: (err: Error) => CanonicalCode = Spanner.getCanonicalCode
-  ): Promise<ReturnT> {
-    const { newOptions, span } = this.createSpan<OptionsT>(options, operationName);
-
-    try {
-      return await fn(newOptions, span);
-    } catch (err) {
-      span.setStatus({
-        code: translateToCanonicalCodeFn(err),
-        message: err.message
-      });
-      throw err;
-    } finally {
-      span.end();
-    }
-  }
-
-  private createSpan<T extends Spannable>(options: T, operationName: keyof TClient) {
-    const span = getTracer().startSpan(`${this.baseOperationName}.${operationName}`, {
-      ...options.spanOptions,
-      kind: SpanKind.INTERNAL
+export async function trace<ReturnT>(
+  operationName: keyof AppConfigurationClient,
+  options: OperationOptions,
+  fn: (options: OperationOptions, span: Span) => Promise<ReturnT>,
+  createSpanFn = createSpan
+): Promise<ReturnT> {
+  const { updatedOptions, span } = createSpanFn(
+    operationName,
+    options
+  );
+
+  try {
+    // NOTE: we really do need to await on this function here so we can handle any exceptions thrown and properly
+    // close the span.
+    return await fn(updatedOptions, span);
+  } catch (err) {
+    span.setStatus({
```

Comment:
Do we need to set a success status if there's no error thrown?

Reply:
Oof, this is interesting.

So in the opentelemetry version we have today the answer is 'no' but that appears to be because the default status code is actually a CanonicalCode, where 0 is equal to CanonicalCode.OK.

In the newer version the default value is StatusCode.UNSET, not OK. If we do need to set StatusCode.OK then we'll have to do something different (and probably, again, for everyone). We can do that piecemeal, and I need to confirm that I'm not just doing something incorrect in my own testing.

Reply:
I changed appconfig to do it. I'll probably need a better solution here for everyone else, but we can take care of this package first.